java-lib/src/main/java/com/yammer/metrics/core/WavefrontHistogram.java

```diff
@@ -61,28 +49,45 @@ public static WavefrontHistogram get(MetricsRegistry registry, MetricName metric
   }
 
   @VisibleForTesting
-  public static synchronized WavefrontHistogram get(MetricsRegistry registry, MetricName metricName, Supplier<Long> clock) {
+  public static synchronized WavefrontHistogram get(MetricsRegistry registry,
+                                                    MetricName metricName,
+                                                    Supplier<Long> clock) {
     // Awkward construction trying to fit in with Yammer Histograms
     TDigestSample sample = new TDigestSample();
     WavefrontHistogram tDigestHistogram = new WavefrontHistogram(sample, clock);
     sample.set(tDigestHistogram);
     return registry.getOrAdd(metricName, tDigestHistogram);
   }
 
-  private WavefrontHistogram(TDigestSample sample, Supplier<Long> millis) {
-    super(sample);
-    this.millis = millis;
-    this.bins = new LinkedList<>();
-  }
+  /**
+   * Aggregates all the bins prior to the current minute
+   * This is because threads might be updating the current minute bin while the bins() method is invoked
+   *
+   * @param clear if set to true, will clear the older bins
+   * @return returns aggregated collection of all the bins prior to the current minute
+   */
+  public Collection<MinuteBin> bins(boolean clear) {
```

Comment:
It's always preferable to return a more specific collection interface as a return type (while allowing the broadest collection interface as inputs). In your case, I would expect it to return a List<>